README.md

```diff
@@ -13,6 +13,223 @@ for kubernetes bootstrap.
+  clusterConfiguration:
+    controllerManager:
+      extraArgs:
+        enable-hostpath-provisioner: "true"
+    networking:
+      podSubnet: "10.244.0.0/16"
+      dnsDomain: ""
+      serviceSubnet: ""
+```
+
+Additional control plane nodes:
+```yaml
+kind: KubeadmConfig
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+metadata:
+  name: my-control-plane2-config
+spec:
+  joinConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+    controlPlane: {}
+```
+
+worker nodes:
+```yaml
+kind: KubeadmConfig
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+metadata:
+  name: my-worker1-config
+spec:
+  joinConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+```
+
+### Simplifying KubeadmConfig objects
+`KubeadmConfig` object allows full control of Kubeadm init/join operations by exposing raw `InitConfiguration`,
+`ClusterConfiguration` and `JoinConfiguration` objects.
+
+> valid combinations of configuration objects are:
+> - at least one of `InitConfiguration` and `ClusterConfiguration` for the first control plane node only
+> - `JoinConfiguration` for worker nodes or additional control plane nodes
+>
+> see [examples](#kubeadmconfig-examples)
+
+By design CABPK respect all the user settings in configuration `KubeadmConfig`; in case some of those settings are missing,
+CABPK applies the following set of rules that can simplify the preparation of `KubeadmConfig`:
+
+- if `KubeadmConfig.clusterConfiguration.clusterName` is empty, CABPK will automatically set this value to
+`Cluster.metadata.name` value.
+- if `KubeadmConfig.clusterConfiguration.controlPlaneEndpoint` is empty, CABPK will automatically set this value to
+the value in the first entry in `Cluster.status.apiEndpoints`; this values is set by the infrastructure provider
+when creating the cluster load balancer or, in some cases, when creating the first control plane machine.
+- if `KubeadmConfig.clusterConfiguration.networking.serviceSubnet` is empty, CABPK will automatically set this value to
+`Cluster.spec.clusterNetwork.serive.cidrBlocks` value. Same apply for the other networking settings.
+- if `KubeadmConfig.joinConfiguration.discovery` is not set, CABPK will automatically create a short lived BootstrapToken
+in your cluster, and then inject corresponding settings into the `JoinConfiguration` object.
+
+### Bootstrap Orchestration
+CABPK orchestrates the generation of cloud-init scripts of different machines in order to ensure a cluster
+bootstrap process that will be compliant with the correct Kubeadm init/join sequence. More in detail:
+1. cloud-init generation starts only after `Cluster.InfrastructureReady` flag is set to `true`.
+2. at this stage, the cloud-init for **one control plane machine only** is generated (kubeadm init).
```

Comment:
It might be nice to say we support multiple control plane machines initing at the same time.

Perhaps don't bold the one control plane machine part and say something like

`the cloud-init script/cloud-config will be generated for the first control plane machine even if multiple control plane machines are ready`